lib/k8s/api/metav1/api_resource.rb

```diff
@@ -5,21 +5,21 @@ module API
     module MetaV1
       # @see https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#APIResource
       class APIResource < Struct
-        attribute :name, Types::String
-        attribute :singularName, Types::String
-        attribute :namespaced, Types::Bool
-        attribute :group, Types::String.optional.default(nil)
-        attribute :version, Types::String.optional.default(nil)
-        attribute :kind, Types::String
-        attribute :verbs, Types::Array.of(Types::Strict::String)
-        attribute(:shortNames, Types::Array.of(Types::Strict::String).optional.default { [] })
-        attribute(:categories, Types::Array.of(Types::Strict::String).optional.default { [] })
+        attribute :name, Types::Strict::String
+        attribute :singularName, Types::Strict::String
+        attribute :namespaced, Types::Strict::Bool
+        attribute :group, Types::Strict::String.optional.default(nil)
+        attribute :version, Types::Strict::String.optional.default(nil)
+        attribute :kind, Types::Strict::String
+        attribute :verbs, Types::Strict::Array.of(Types::Strict::String)
+        attribute :shortNames, Types::Strict::Array.of(Types::Strict::String).optional.default([])
```

Comment:
These default arrays can mess things up since the same array instance is shared with all struct instances.. I think the old dry took procs like `.default(proc { [] })`.

Reply:
Yes, there are such in the following files.

Reply:
Fixed.